LiteCore/Database/DatabaseImpl.cc

```diff
@@ -576,12 +576,29 @@ namespace litecore {
                 DebugAssert(isValidScopeNameOrDefault(scope));
                 name.setStart(slash + 1);
             }
+            // CBL-3824: This is necessary for determining if the keystore is an index, which may occur in
+            // the case of a Full-Text Index. The keystore name for an FTI looks like: <scope>.<collection>::<index>
+            slice idxName = nullslice;
+            // Find the first ':'
+            if (auto idxSep = name.findByte(KeyStore::kIndexSeparator)) {
+                idxName = slice(idxSep+1, name.end());
+                // Find the second colon and set index name accordingly
```

Comment:
Is it necessary to find the second colon?  In your example it just seems to be one position after the first.  If that's the case then the second find is overkill and you can just check the following byte, or better yet use find("::"_sl)

Reply:
Thanks Jim 🙂 I wasn't sure on the best solution, but this didn't feel like it. Yours is a much better one.